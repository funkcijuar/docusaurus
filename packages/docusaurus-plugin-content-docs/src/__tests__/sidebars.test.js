/**
 * Copyright (c) 2017-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import loadSidebars from '../sidebars';

/* eslint-disable global-require, import/no-dynamic-require */

describe('loadSidebars', () => {
  test('sidebars with known sidebar item type', async () => {
    const sidebarPath = path.join(
      __dirname,
      '__fixtures__',
      'website',
      'sidebars.json',
    );
    const result = loadSidebars(sidebarPath);
    expect(result).toMatchSnapshot();
  });

  test('sidebars with unknown sidebar item type', async () => {
    const sidebarPath = path.join(
      __dirname,
      '__fixtures__',
      'website',
      'bad-sidebars.json',
    );
    expect(() => loadSidebars(sidebarPath)).toThrowErrorMatchingInlineSnapshot(
      `"Unknown sidebar item type: superman"`,
    );
  });

  test('no sidebars', () => {
    const result = loadSidebars(null);
    expect(result).toEqual({});
  });
});
