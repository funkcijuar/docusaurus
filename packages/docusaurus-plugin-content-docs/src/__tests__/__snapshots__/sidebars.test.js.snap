// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadSidebars sidebars with known sidebar item type 1`] = `
Object {
  "docs": Array [
    Object {
      "items": Array [
        Object {
          "id": "foo/bar",
          "type": "doc",
        },
        Object {
          "id": "foo/baz",
          "type": "doc",
        },
        Object {
          "href": "https://github.com",
          "label": "Github",
          "type": "link",
        },
        Object {
          "id": "hello",
          "type": "ref",
        },
      ],
      "label": "Test",
      "type": "category",
    },
    Object {
      "items": Array [
        Object {
          "id": "hello",
          "type": "doc",
        },
      ],
      "label": "Guides",
      "type": "category",
    },
  ],
}
`;
